src/noit_ssl10_compat.h

```diff
@@ -0,0 +1,31 @@
+#ifndef NOIT_SSL10_COMPAT_H
+#define NOIT_SSL10_COMPAT_H
+
+#include <openssl/opensslv.h>
+
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+
+#include <string.h>
+#include <openssl/engine.h>
+
+inline void *OPENSSL_zalloc(size_t num)
+{
+   void *ret = OPENSSL_malloc(num);
+
+   if (ret != NULL)
+       memset(ret, 0, num);
+   return ret;
+}
+
+inline EVP_MD_CTX *EVP_MD_CTX_new(void)
+{
+  return OPENSSL_zalloc(sizeof(EVP_MD_CTX));
+}
+
+inline void EVP_MD_CTX_free(EVP_MD_CTX *ctx)
+{
+  EVP_MD_CTX_cleanup(ctx); OPENSSL_free(ctx);
```

Comment:
please use two lines, no reason to be on a single line

Reply:
this is copied from libmtev's version of this header

Reply:
FWIW both libmtev's version and this were just copied from OpenSSL's migration guide.